packages/jest-config/src/index.ts

```diff
@@ -267,6 +229,92 @@ This usually means that your ${chalk.bold(
   }
 };
 
+export interface ReadJestConfigOptions {
+  /**
+   * The package root or deserialized config (default is cwd)
+   */
+  packageRootOrConfig?: string | Config.InitialOptions;
+  /**
+   * When the `packageRootOrConfig` contains config, this parameter should
+   * contain the dirname of the parent config
+   */
+  parentConfigDirname?: null | string;
+  /**
+   * Indicates whether or not to read the specified config file from disk.
+   * When true, jest will read try to read config from the current working directory.
+   * (default is false)
+   */
+  readFromCwd?: boolean;
+  /**
+   * Indicates whether or not to ignore the error of jest finding multiple config files.
+   * (default is false)
+   */
+  skipMultipleConfigError?: boolean;
+}
+
+/**
+ * Reads the jest config, without validating them or filling it out with defaults.
+ * @param config The path to the file or serialized config.
+ * @param param1 Additional options
+ * @returns The raw initial config (not validated)
+ */
+export async function readInitialOptions(
+  config?: string,
+  {
+    packageRootOrConfig = process.cwd(),
+    parentConfigDirname = null,
+    readFromCwd = false,
+    skipMultipleConfigError = false,
+  }: ReadJestConfigOptions = {},
+): Promise<{config: Config.InitialOptions; configPath: string | null}> {
+  if (typeof packageRootOrConfig !== 'string') {
+    if (parentConfigDirname) {
+      const rawOptions = packageRootOrConfig;
+      rawOptions.rootDir = rawOptions.rootDir
+        ? replaceRootDirInPath(parentConfigDirname, rawOptions.rootDir)
+        : parentConfigDirname;
+      return {config: rawOptions, configPath: null};
+    } else {
+      throw new Error(
+        'Jest: Cannot use configuration as an object without a file path.',
+      );
+    }
+  }
+  if (isJSONString(config)) {
+    // A JSON string was passed to `--config` argument and we can parse it
+    // and use as is.
+    let initialOptions;
+    try {
+      initialOptions = JSON.parse(config);
+    } catch {
+      throw new Error(
+        'There was an error while parsing the `--config` argument as a JSON string.',
+      );
+    }
+
+    // NOTE: we might need to resolve this dir to an absolute path in the future
+    initialOptions.rootDir = initialOptions.rootDir || packageRootOrConfig;
+    return {config: initialOptions, configPath: null};
+    // A string passed to `--config`, which is either a direct path to the config
+    // or a path to directory containing `package.json`, `jest.config.js` or `jest.config.ts`
```

Comment:
This comment should be moved two lines down. Or?

Reply:
You are right! Fixed